flowmachine/flowmachine/core/cache.py

```diff
@@ -430,22 +455,35 @@ def shrink_below_size(
         Optionally override the maximum cache size set in flowdb.
     dry_run : bool, default False
         Set to true to just report the objects that would be removed and not remove them
+    protected_period : int, default None
+        Optionally specify a number of seconds within which cache entries are excluded. If None,
+        the value stored in cache.cache_config will be used. Set to -1 to ignore cache protection
+        completely.
 
     Returns
     -------
     list of "Query"
         List of the queries that were removed
     """
     initial_cache_size = get_size_of_cache(connection)
+    if size_threshold is None:
+        size_threshold = get_max_size_of_cache(connection)
     removed = []
     logger.info(
-        f"Shrinking cache from {initial_cache_size} to below {size_threshold}{' (dry run)' if dry_run else ''}."
+        f"Shrinking cache from {initial_cache_size} to below {size_threshold}{' (dry run)' if dry_run else ''}.",
+        initial_cache_size=initial_cache_size,
+        size_threshold=size_threshold,
+        dry_run=dry_run,
     )
 
     if dry_run:
```

Comment:
I'm not sure dry_run will work correctly if a 'wet' run would remove more than one cached query - looks like a dry run would repeatedly 'remove' the same query at each iteration. There's a test to ensure that dry and wet runs give the same results, but in that test only one query is removed from cache.

I can't see a straightforward fix, though, and this issue isn't directly related to this PR, so I'm happy to make an issue and leave it for now if you'd prefer.

Reply:
No, I read this a bit too quickly. Looks like it should work, but I don't think the test is checking this because it's only removing one query.